test/api_deployment/main.tf

```diff
@@ -39,9 +39,16 @@ module "lambdas" {
 
   lambda_file = "lambda.zip"
   functions = { ApiDeploymentTestLambda = { handler = "lambda.handler" }}
-  source_arn = "arn:aws:execute-api:${var.region}:${data.aws_caller_identity.current.account_id}:${aws_api_gateway_rest_api.api.id}/*/GET/"
-  statement_id = "AllowExecutionFromAPIGateway"
-  principal = "apigateway.amazonaws.com"
+
+  permission_count = 1
+  permissions = [
+    {
+      principal    = "apigateway.amazonaws.com"
+      statement_id = "AllowExecutionFromAPIGateway"
+      source_arn   = "arn:aws:execute-api:${var.region}:${data.aws_caller_identity.current.account_id}:${aws_api_gateway_rest_api.api.id}/*/GET/"
+    }
```

Comment:
This will only allow permissions for GET requests, right? Do we want to make this a little more generic now? Or create an issue to get round to it at some point? I think we have other places in the modules where GET is hard coded.